src/data_structures/stack.rs

```diff
@@ -0,0 +1,158 @@
+use std::fmt::{self, Display, Formatter};
+use std::ptr;
+use std::ptr::NonNull;
+
+struct Node<T> {
+    val: T,
+    next: Option<NonNull<Node<T>>>,
```

Comment:
Why not just use `Option<Box<Node<T>>>`?

Reply:
very good point. I will change that.